packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -58,6 +58,9 @@ export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveM
   ComposedComponent: new (props: TProps, ...args: any[]) => React.Component<TProps, TState>,
 ): any {
   const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
+    public static contextType = WindowContext;
+    public context: React.ContextType<typeof WindowContext>;
```

Comment:
Must these be `public`?

Reply:
Yes, React expects it to be public (like the component lifecycles) https://reactjs.org/docs/context.html#classcontexttype

Reply:
:( Yeah I found that out. I was saddened by it too.

Reply:
Possible to use `Context.Consumer` as an alternative. Maybe there are reasons why you didn't though.

```jsx
  <WindowContext.Consumer>{window => (
            <.../>
   )}</UserContext.Consumer>
```

Reply:
I had it that way initially, but there were concerns about it rerendering too often. The alternative was to make another child component but that would increase the dom tree size and complexity.